File: Users/Adam.java
public class Adam extends Room15User {

	private float speed = 0.0;
	private float ping = Float.MAX_VALUE; // Expressed in ms.
	private boolean isRoomOwner = true;
	private boolean isDumbass = true;
	private boolean hasCircle = true;

	// TODO Create method to find keystore.
	private KeyStore ks;

	//Number of times Adam has lost his keystore
	int keystoreLost = 1;

	private ArrayList<DadJoke> dadJokes;

	private ArrayList<String> tristanRandomLinksBuffer;

	public Adam() {
		ks = null;
		dadJokes = new ArrayList<>();
		tristanRandomLinksBuffer = new ArrayList<>();
		dadJokes.ensureCapacity(Integer.MAX_VALUE);
		tristanRandomLinksBuffer.ensureCapacity(Integer.MAX_VALUE);
	}

	//Always returns true because Adam is an idiot and lost his keystore
	public boolean hasLostKeystore(){
		return true;
	}

	public boolean isCircle(){
		return hasCircle;
	}

	public boolean isOwner(){
		return isRoomOwner;
	}
	
	//Method that returns the number of times Adam has lost his keystore
	public int keystoreLostCount(){
		return keystoreLost;
	}


	public ArrayList<String> inLoveWith(){
		//Initialize list of lovers
    	ArrayList lovers = new ArrayList<String>();

    	//Add lovers in order from most in love to least
    	lovers.add("Tristan");
    	lovers.add("Mauker");
    	lovers.add("Karley");
    

    	//return lovers
		return lovers;
	}

	@Override
	public Object clone(){
		// Soon that will be true.
		return new сodeMagic();
	}
}

File: Users/Adam.kt
class Adam : Room15User {
	private val speed = 1.0F // He's getting faster. http://chat.stackoverflow.com/transcript/message/30989844#30989844
	private val ping = Float.MAX_VALUE // Expressed in MS
	private val isDumbass = true
	private val hasAbandonedUs = true
	private val dadJokes: List<DadJoke>
	private val tristanRandomLinksBuffer: List<String>

	//TODO: Create method to find keystore
	private val ks: KeyStore?

	init {
		ks = null
		dadJokes = ArrayList(Integer.MAX_VALUE)
		tristanRandomLinksBuffer = ArrayList(Integer.MAX_VALUE)
	}

	val hasLostKeystore: Boolean = true
	val hasCircle = true
	val isRoomOwner = true
	var keystoreLostCount = 1

	// Add lovers in order from most to least.
	val lovers: List<String> = listOf("Tristan", "Mauker")

	override fun clone() = codeMagic()
}